### ISSUE TITLE
Apk doesn't have --no-cache option on Alpine 3.1

### ISSUE THREAD

**Issue by Andy-ch** [Andy-ch/docker-nocache#3]
```
2020-06-20T12:18:15.6925133Z + docker buildx build --platform linux/amd64 -t localhost:5000/andych/alpine-nocache:3.1 --build-arg tag=3.1 --push .
2020-06-20T12:18:15.9324259Z #2 [internal] load .dockerignore
2020-06-20T12:18:15.9326940Z #2 transferring context: 2B done
2020-06-20T12:18:15.9328217Z #2 DONE 0.0s
2020-06-20T12:18:15.9329121Z 
2020-06-20T12:18:15.9330940Z #1 [internal] load build definition from Dockerfile
2020-06-20T12:18:15.9331497Z #1 transferring dockerfile: 31B done
2020-06-20T12:18:15.9331795Z #1 DONE 0.0s
2020-06-20T12:18:15.9331914Z 
2020-06-20T12:18:15.9332744Z #3 [internal] load metadata for localhost:5000/andych/alpine-nocache:3.1
2020-06-20T12:18:15.9332927Z #3 DONE 0.0s
2020-06-20T12:18:15.9333205Z 
2020-06-20T12:18:15.9333715Z #4 [1/2] FROM localhost:5000/andych/alpine-nocache:3.1@sha256:1a23a8f715d26...
2020-06-20T12:18:15.9334291Z #4 resolve localhost:5000/andych/alpine-nocache:3.1@sha256:1a23a8f715d26c23df4cf8cfe9d3ca05b978dd4131fa54634190156bc03583c9 done
2020-06-20T12:18:15.9334467Z #4 CACHED
2020-06-20T12:18:15.9334550Z 
2020-06-20T12:18:15.9334860Z #5 [2/2] RUN apk add curl &&     exit "$(ls /var/cache/apk/|wc -l)"
2020-06-20T12:18:15.9335160Z #5 0.062 unrecognized option `--no-cache'
2020-06-20T12:18:15.9335325Z #5 0.066 WARNING: Ignoring APKINDEX.a9892af1.tar.gz: No such file or directory
2020-06-20T12:18:15.9335691Z #5 0.067 ERROR: unsatisfiable constraints:
2020-06-20T12:18:15.9335835Z #5 0.067   curl (missing):
2020-06-20T12:18:15.9335976Z #5 0.067     required by: world[curl]
2020-06-20T12:18:15.9339844Z #5 ERROR: executor failed running [/bin/sh -c apk add curl &&     exit "$(ls /var/cache/apk/|wc -l)"]: buildkit-runc did not terminate successfully
```

**Comment by Andy-ch** [comment]
Solution: create different apk executable substitutes. One with --no-cache option, for newer version, another with simply clearing cache dir after running apk-original. Run special executable in Dockerfile which will determine current Alpine version and decide which substitute executable to use, and which one to delete

**Comment by Andy-ch** [comment]
Resolved in #4
